services/provider/proto/provider.proto

```diff
@@ -109,6 +118,21 @@ message StorageConfigResponse{
     SystemAttributes systemAttributes = 3;
 }
 
+message ClientAttributes {
+  bool inMaintenanceMode = 1;
+  bool mirrorEnabled = 2;
+  string desiredSubscriptionChannel = 3;
+}
+
+// StorageInfoResponse holds the response for the GetStorageInfo API request
+message StorageInfoResponse{
+  // ExternalResource holds the configuration data of external storage cluster
+  repeated bytes kubeResources = 1;
+  // Contains hash of desired config
+  string desiredConfigHash = 2;
+  ClientAttributes clientAttributes = 3;
```

Comment:
Now that everything is typed, is there really a need for this indirection?

Reply:
Do we need the clientAttributes to be directly embedded into the GetDesiredClientStateResponse?